sklearn/ensemble/voting.py

```diff
@@ -339,9 +369,157 @@ def get_params(self, deep=True):
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.ensemble import RandomForestRegressor, AverageRegressor
+    >>> r1 = LinearRegression()
+    >>> r2 = RandomForestRegressor(n_estimators=10, random_state=1)
+    >>> X = np.array([[1, 1], [2, 4], [3, 9], [4, 16], [5, 25], [6, 36]])
+    >>> y = np.array([2, 6, 12, 20, 30, 42])
+    >>> er = AverageRegressor(estimators=[
+    ...         ('lr', r1), ('rf', r2)])
+    >>> print(er.fit(X, y).predict(X))
+    [ 3.3  5.7 11.8 19.7 28.  40.3]
+    """
+
+    def __init__(self, estimators, weights=None, n_jobs=None):
+        self.estimators = estimators
+        self.weights = weights
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y, sample_weight=None):
+        """ Fit the estimators.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        sample_weight : array-like, shape = [n_samples] or None
+            Sample weights. If None, then samples are equally weighted.
+            Note that this is supported only if all underlying estimators
+            support sample weights.
+
+        Returns
+        -------
+        self : object
+        """
+        return super(AverageRegressor, self).fit(X, y, sample_weight)
+
+    def predict(self, X):
+        """Predict regression target for X.
+
+        The predicted regression target of an input sample is computed as the
+        mean predicted regression targets of the estimators in the ensemble.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape = [n_samples, n_features]
+            The training input samples. Sparse matrices are accepted only if
+            they are supported by the base estimator.
+
+        Returns
+        -------
+        y : array of shape = [n_samples]
+            The predicted values.
+        """
+        check_is_fitted(self, "estimators_")
+        return np.average(self._predict(X), axis=1,
+                          weights=self._weights_not_none)
+
+    def transform(self, X):
+        """Return predictions for X for each estimator.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        Returns
+        -------
+        predictions
+            array-like of shape (n_samples, n_classifiers), being
+            values predicted by each regressor.
+        """
+        check_is_fitted(self, 'estimators_')
+        return self._predict(X)
+
+    def set_params(self, **params):
```

Comment:
why does this and `get_params` need redefinition?

Reply:
docstrings for voting and averaging is different.

Reply:
Very few if any set_params implementations say the name of the class in their docstring.. Just change the docstring to generic

Reply:
done.